### DISCUSSION TITLE
Place the footer after the border of master page.

### DISCUSSION THREAD

**Discussion by tnation1424** [eclipse-birt/birt#1457] [Q&A]
Hello there,
  I’m hoping anyone can help me.  Is there a way to place the footer after the border of master page. The problem is that Left and Right borders are running pass the Footer border and the requirement needs those border to end at the footer.  Of course Master Page is only doing what it coded to do.  Is there a way around this?  Having the border on the Layout doesn't work for times when the information ends and there is a big gap until the footer is displayed. Any suggestions with be great thank you.

![image](https://github.com/eclipse-birt/birt/assets/149100941/1e860acf-d70a-4a3e-8671-f986a469f00c)

![image](https://github.com/eclipse-birt/birt/assets/149100941/62558f8c-b00b-4e22-8718-8495bb8b56aa)

**Comment by speckyspooky** [comment]
This is realy hard buth currently there is no option to draw the footer behind the master-page because the drwa-area of the emitter will be defined in whole through the master size and the master-border shows the maximum of all.

**Comment by speckyspooky** [comment]
The only way would be that you use a grid which hase your page height or you define a table like a wrapping table where the rows will have the size of the page then you can create the require border. Then you can add the your content into the simulated page cell but you have to handle on your own way the page handling because the row-height are fixed.

But maybe there is a very easy way instead to use the master-page border.
My suggestion would be to create a background image like a transparent PNG with your border and add this to the report
With BIRT 4.13 you should be able to add the background-image like embedded image but the way which works also before was the usage of the image like external reference via URL.

**Comment by tnation1424** [comment]
Thank you for confirming what I thought had to done.  Making a image as the background for Master Page works great when the report is exported as a PDF! However, it the BIRT viewer it's looks like a hot mess.  

This image is from the BIRT Viewer where that line is the image bleeding over into the body of the report layout.

![image](https://github.com/eclipse-birt/birt/assets/149100941/36ec6f6c-f7bf-4d58-9cbe-10d0c64d30ad)

The second image is from the exported pdf of the report.  Looks perfect!
![image](https://github.com/eclipse-birt/birt/assets/149100941/4951f2e3-8f2d-4061-bef7-f7737b9b3294)

Here's the advanced Master Page Property setting for the Background image. 
![image](https://github.com/eclipse-birt/birt/assets/149100941/842b62e8-1e05-44a5-9760-88214193e68a)


I've tried number of ways and nothing has worked.  Any help will be appreciated, thank you!

**Reply by speckyspooky** [reply]
If you send an example report together with your borders I can take a look into it.
May be your image use a kind of space from the lefthand side to the left border and the html preview draw this range
and therefore you have this overlapping.

**Comment by tnation1424** [comment]
I'm sorry for not responding sooner I try to fix the problem based off your advice and I'm having no success.  I'm missing something, so I'm attaching images to help to see what is possible solution.

Here is the image of how the report looks like in BIRT Viewer.  The line on the left is from the background Image, and notice there isn't a right border.  To me looks like the view is left justifying the layout, while the Master Page is treated differently,
 
![image](https://github.com/eclipse-birt/birt/assets/149100941/ed13531f-390b-4c57-ae10-87d609231e22)

Below is the image of pdf export of the same report.  It looks perfect! This is how I want to look like in the BIRT Viewer.
![image](https://github.com/eclipse-birt/birt/assets/149100941/d2a9ae54-a919-427c-89b3-e365c723768d)

Here is what the Master Page looks like.
![image](https://github.com/eclipse-birt/birt/assets/149100941/87af42bc-533c-45e6-976e-2ebd0e09afd4)

Below are the settings for the Master Page that I have it so far.  I have tried to move the Image to the left by decreasing the Left Margin, but that only messes up the pdf export even more.  If more information is need from me please let me know.  Again I truly appreciate.

![image](https://github.com/eclipse-birt/birt/assets/149100941/259fe70f-9b24-4271-859c-1ae72512b180)

![image](https://github.com/eclipse-birt/birt/assets/149100941/3e94ba65-e866-400f-aaa6-6ee9380c2de1)

![image](https://github.com/eclipse-birt/birt/assets/149100941/1a237ea3-6de7-48b9-b5d3-6afb5c028bf9)

**Reply by speckyspooky** [reply]
My target was to get your demo report and your used image to play a little bit with it.
It could be that you have a transparent area on your PNG-image which cause the movement in the BIRT-viewer.
The viewer doesn't display in the same way the master-page in all details. So the transparent image area on lefthand side will cause the effect.

**Comment by tnation1424** [comment]
Again, I'm sorry for not responding back sooner, busy.  For security reasons I can't share the report.  Is there a way of making the background image on the Master Page visible only when the report is exported to PDF? Again, thank you for any help you can provide.

**Reply by hvbtup** [reply]
You can try setting the CSS property (see auto-completion on `this.getStyle().`) in the `onCreate` event dynamically depending on the output format.

**Comment by tnation1424** [comment]
Last week speckyspooky asked if I could provide a demo of the report to play with it for finding a solution. Today a got the approval to share it.  Again thank you for any help that can be provided.

**Reply by speckyspooky** [reply]
For some testing a small simplified version would be enough.
And if you wants to fix only the HTML-preview because you mentioned the PDF looks good
then what would be your target to remove this border or try to create a better position.
My favorit would be to avoid the border because your requirement is a combination of border and footer (so it makes no sense to have only 1 part of it).
